### DISCUSSION TITLE
Should Table have different name or a small disclaimer

### DISCUSSION THREAD

**Discussion by bq-wrongway** [fyne-io/fyne#3570] [General]
Hi All, 
i wanted just to address on how current Table term is bit deceptive, and find best way to make clear what table in fyne is and how it acts.
I wont talk much about underlying mechanisam of the table, but rather about visual representation, and visual feedback from the table.

To get started lets take a look at table in some other GUI frameworks, like javafx or flutter.
Both have header (fyne does not have it atm but there is a plan to add it)
Both look at a rows as a same TYPE with different value ( row 1 has user1, row 2 has user 2)
Clicking on a row, highlights the whole row, telling us that this in fact is a single item 

Fyne Table by all metrics of visual representation is more of a spreadsheet then it is an table.

Checking for example excel spreadsheet or ControlsFX spreadsheetview, we can see that Fyne table is much closer to them in terms of visual representation, then to the javafx / flutter /swing table.

Again, fyne table in its current form has its uses and its very cool, but its not what typical user expects from a table.

Maybe adding small note to the documentation page would clear this up, and maybe in future it would make sense to have fyne DataTable that would look and act  more in line  with what user expects from the table.

Again this is neither a complaint, or a request, im simply trying to point out how term "Table" in fyne gives us something that is not exactly what we have come to expect from a table.

**Comment by andydotxyz** [comment]
This feels like a duplicate of the request for adding headers. I'm not sure what else is needed from the text above?

**Reply by bq-wrongway** [reply]
Hi sorry for the late reply, its not really request for headers, fyne table is missing some other characteristic that tables usually have, it would be ok just that there is some heads up on the docu page.
(no headers at the moment, no sorting, no way to highlight whole row) coming from any other GUI framework this can be confusing.

**Reply by andydotxyz** [reply]
It depends what you are coming from. Some tables are just layout with headers (like HTML) some don't exist at all (iOS).
So, no, I don't think we need a better name or a disclaimer - however if you would like to open a feature request for items missing that would help us to move forward.

**Reply by bq-wrongway** [reply]
Thank you for your answer,i will think about this in make a feature request in following days. This discussion can be closed.

**Reply by Jacalz** [reply]
FYI: For headers specifically, there is an issue open already at https://github.com/fyne-io/fyne/issues/1658 :)

**Comment by junmaqiang** [comment]
yes,i think tables need th too
